From 5b010aa1fa954855ad67788f0a3a35f6805e9c21 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Tue, 28 Jul 2020 23:48:41 +0200
Subject: [PATCH] Also call requestHideAutoHidingPanel for panels set to
 "windows can cover", so that the screen edge gets recreated for it

---
 shell/panelview.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/shell/panelview.cpp b/shell/panelview.cpp
index ddf6b0b70..0f17260ea 100644
--- a/shell/panelview.cpp
+++ b/shell/panelview.cpp
@@ -746,11 +746,12 @@ void PanelView::setAutoHideEnabled(bool enabled)
         KWindowEffects::slideWindow(winId(), slideLocation, -1);
     }
 #endif
-    if (m_shellSurface && m_visibilityMode == PanelView::AutoHide) {
+    if (m_shellSurface && (m_visibilityMode == PanelView::AutoHide || m_visibilityMode == PanelView::LetWindowsCover)) {
         if (enabled) {
             m_shellSurface->requestHideAutoHidingPanel();
         } else {
-            m_shellSurface->requestShowAutoHidingPanel();
+            if (m_visibilityMode == PanelView::AutoHide)
+                m_shellSurface->requestShowAutoHidingPanel();
         }
     }
 }
-- 
GitLab

